src/transformStateWithClones.js

```diff
@@ -7,7 +7,46 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const options = Object.values(actions);
+  const result = [];
+
+  for (const option of options) {
+    const { keysToRemove, type, extraData } = option;
+    const stateCopy = { ...state };
```

Comment:
why you need to create a state copy on each loop iteration? do it outside of loop once